ai/ai_model_building/stat_parser.py

```diff
@@ -0,0 +1,53 @@
+import re
+from model import Stats
+
+stats_config = {
+    'global': ['Ad', 'Ma', 'Di'],
+    'defense': ['HP', 'Df', 'Re', 'Arm'],
+    'offense': ['At', 'Of', 'St', 'AP', 'Ag']
+}
+
+stat_model_param = ['advance', 'march', 'discipline', 'health_points', 'defense',
+                    'resilience', 'armour', 'attack', 'offensive', 'strength', 'armour_penetration', 'agility']
+
+
+def handle_special_values(value):
+    def replace_dice_six(match): return str(
+        int(match.group(1)) * 3) if match.group(1) else '3'
+    def replace_dice_three(match): return str(
+        int(match.group(1)) * 2) if match.group(1) else '2'
+    value = value.replace('C', '0')
+    value = re.sub('(\d)*D6', replace_dice_six, value)
+    value = re.sub('(\d)*D3', replace_dice_three, value)
+    return value
```

Comment:
Why 2 nested functions ? I think it's a code smell

Reply:
I needed those function right only right under and the only two options I had where lambda functions and nested functions.
As you can see, the functions contains a condition. I definitely think it's more readable that way rather than with lambdas